compute_tools/src/config.rs

```diff
@@ -157,3 +159,82 @@ where
 
     res
 }
+
+fn get_rsyslog_pid() -> Option<String> {
+    let output = Command::new("pgrep")
+        .arg("rsyslogd")
+        .output()
+        .expect("Failed to execute pgrep");
+
+    if !output.stdout.is_empty() {
+        let pid = std::str::from_utf8(&output.stdout)
+            .expect("Invalid UTF-8 in process output")
+            .trim()
+            .to_string();
+        Some(pid)
+    } else {
+        None
+    }
+}
+
+// Start rsyslogd with the specified configuration file
+// If rsyslogd is running and restart it if necessary
+fn restart_rsyslog(rsyslog_conf_path: &str) -> Result<()> {
+    let pid = get_rsyslog_pid();
+    if let Some(pid) = pid {
+        info!("rsyslogd is already running with pid: {}", pid);
+        // kill the running rsyslogd
+        let _ = Command::new("kill").arg("-9").arg(&pid).output();
```

Comment:
Will `kill -9` instantly kill rsyslog? I am worried that when we want to reconfigure rsyslog on the fly we then might lose some data.

Reply:
I changed the logic to not restart rsyslog if it is already running, because we don't need online reconfiguration for hipaa use-case.

-HUP signal doesn't reload rsyslog config, it only affects the log file itself:
https://github.com/rsyslog/rsyslog/blob/1280ae4d1d2673636d073e6991801250c565cdf2/tools/rsyslogd.8#L154